liblouis/lou_translateString.c

```diff
@@ -2206,33 +2206,36 @@ for_selectRule(const TranslationTableHeader *table, int pos, OutString output, i
 static int
 undefinedCharacter(widechar c, const TranslationTableHeader *table, int pos,
 		const InString *input, OutString *output, int *posMapping, int *cursorPosition,
-		int *cursorStatus) {
+		int *cursorStatus, int mode) {
 	/* Display an undefined character in the output buffer */
-	int k;
-	char *display;
-	widechar displayDots[20];
 	if (table->undefined) {
 		TranslationTableRule *rule =
 				(TranslationTableRule *)&table->ruleArea[table->undefined];
-		if (!for_updatePositions(&rule->charsdots[rule->charslen], rule->charslen,
-					rule->dotslen, 0, pos, input, output, posMapping, cursorPosition,
-					cursorStatus))
-			return 0;
-		return 1;
+
+		return 	for_updatePositions(&rule->charsdots[rule->charslen], rule->charslen,
+					    rule->dotslen, 0, pos, input, output, posMapping, cursorPosition,
+					    cursorStatus);
 	}
-	display = _lou_showString(&c, 1);
-	for (k = 0; k < (int)strlen(display); k++)
-		displayDots[k] = _lou_getDotsForChar(display[k]);
-	if (!for_updatePositions(displayDots, 1, (int)strlen(display), 0, pos, input, output,
-				posMapping, cursorPosition, cursorStatus))
-		return 0;
-	return 1;
+
+	const char *text = (mode & noUndefined)? "": _lou_showString(&c, 1, 1);
```

Comment:
We normally put variable declarations at the beginning of the block. Not sure if this has changed now. @egli?

Reply:
C99 allows to define vars where they are needed. I always liked to keep the definition as close to the use as possible. The "legacy" code of course keeps the declarations at the beginning of the block, but I am certainly not advocating that.

Reply:
OK great. Then from now there's no reason to keep declaring the variables at the top.

Reply:
> C99 allows to define vars where they are needed. I always liked to keep the definition as close to the use as possible. The "legacy" code of course keeps the declarations at the beginning of the block, but I am certainly not advocating that.

The main reason I still define all variables at the beginning without defining them is that it makes gdb debugging scripts easier to write and maintain.